File: api/src/main/java/tutmirleid/api/entity/PageEntity.java
package tutmirleid.api.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;

@Entity
@Table(name = "page")
public class PageEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Long id;

    @Column
    private String link;

    @Column
    private String title;

    @Column
    private String alias;

    public PageEntity() {}

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAlias() {
        return alias;
    }

    public void setAlias(String alias) {
        this.alias = alias;
    }
}

File: api/src/entity/page.entity.ts
import {Column, Entity, PrimaryGeneratedColumn} from "typeorm";

@Entity({ name: 'page' })
export class PageEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({
        nullable: false,
        type: 'varchar',
        length: 255
    })
    title: string;

    @Column({
        nullable: false,
        type: 'varchar',
        length: 255
    })
    link: string;

    @Column({
        nullable: false,
        type: 'varchar',
        length: 255
    })
    alias: string;
}